src/components/UsedByItem.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { FC } from "react";
+import ResourceLink from "./ResourceLink";
+import { LxdUsedBy } from "util/usedBy";
+import { ResourceIconType } from "./ResourceIcon";
+import { useQuery } from "@tanstack/react-query";
+import { fetchImage } from "api/images";
+import { queryKeys } from "util/queryKeys";
+
+interface Props {
+  item: LxdUsedBy;
+  project: string;
+  type: ResourceIconType;
+  to: string;
+}
+
+const UsedByItem: FC<Props> = ({ item, project, type, to }) => {
+  const { data: image } = useQuery({
+    queryKey: [queryKeys.images, item.name],
+    queryFn: () => fetchImage(item.name, project),
+    enabled: type === "image",
+  });
+
+  let label = item.name;
+  if (type === "image") {
+    label = image?.properties?.description || item.name;
+  }
+
+  const fromAnotherProject = item.project !== project;
+  const actualProject = fromAnotherProject ? item.project : project;
+  return (
+    <div>
+      {fromAnotherProject && (
+        <>
+          <ResourceLink
+            type="project"
+            value={actualProject}
+            to={`/ui/project/${actualProject}/instances`}
+          />{" "}
+          /{" "}
+        </>
+      )}
+      {type === "snapshot" && (
+        <>
+          <ResourceLink
+            type={item.instance ? "instance" : "volume"}
+            value={item.instance || item.volume || ""}
+            to={
+              item.instance
+                ? `/ui/project/${actualProject}/instance/${item.instance}`
+                : `/ui/project/${actualProject}/storage/pool/${item.pool}/volumes/custom/${item.volume}`
+            }
+          />{" "}
+          /{" "}
+        </>
+      )}
```

Comment:
I think branching out the instance/volume cases on a higher level, would make this more explicit and simpler to read. So basically `{type === "snapshot" && item.instance && (...)` and a second similar case for the volume below.